### ISSUE TITLE
Horizontal scrolling with scroll wheel on Logitech MX Master 3 doesn't work

### ISSUE THREAD

**Issue by marcusthorman** [notepad-plus-plus/notepad-plus-plus#10490]
### Description of the Issue

When scrolling with horizontal scroll wheel on the Logitech MX Master 3, the scrollbar indicates that you have scrolled, but no scrolling actually occurs.

https://user-images.githubusercontent.com/57597060/131803900-318f84ca-fb1b-4dd1-a911-34b4f4a6dd6b.mp4

### Steps to Reproduce the Issue

1.  Open document in npp
2.  Scroll horizontally with scrollwheel

### Expected Behavior

Scrolling horizontally with the scrollwheel should scroll in the file

### Actual Behavior

Scrollbar indicates that you have scrolled, but the view of the file does not update accordingly

### Debug Information

Notepad++ v8.1.3   (32-bit)
Build time : Aug 10 2021 - 00:32:53
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Home (64-bit) 
OS Version : 2009
OS Build : 19043.1165
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer.dll

**Comment by rimrul** [comment]
Probably related to, if not a duplicate of #8794 and #2096

**Comment by dail8859** [comment]
For me the scroll wheel works (though it does let me scroll past the limits of the scrollbar). So not sure if it is some configuration in Logitech Options (which I do have installed) or some setting in Notepad++

```
Notepad++ v8.1.4   (64-bit)
Build time : Aug 21 2021 - 13:04:59
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1909
OS Build : 18363.1734
Current ANSI codepage : 1252
Plugins : BetterMultiSelection.dll ComparePlugin.dll ElasticTabstops.dll LuaScript.dll mimeTools.dll NppConverter.dll NppExport.dll NppMarkdownPanel.dll SurroundSelection.dll XMLTools.dll 
```

**Comment by dail8859** [comment]
Literally right after I posted that comment, I `alt+tabbed` back to Notepad++ and now have same issue you posted. 🤣

**Comment by ozone10** [comment]
Seems more like issue with Logitech Options, IIRC not only Notepad++ but other applications have this issue.

Until Logitech fix it workaround is to not use Logitech Options, or run Notepad++ as admin.

**Comment by EscolarProgramming** [comment]
Same problem here 

```
Notepad++ v8.2.1   (64-bit)
Build time : Jan 19 2022 - 18:43:05
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 (64-bit) 
OS Version : 2009
OS Build : 22000.434
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll 
```

**Comment by nnguy012** [comment]
Hi everyone, this solution worked for me...
I noticed the horizontal scrolling works fine when you run notepad++ in admin, obviously enabling UAC everytime is annoying...

So... I uninstalled notepad++ and reinstalled it in a location that does not require UAC intervention, in my case I installed it in C:\Users\<username>\Notepad++

Works every time now.

Just FYI:
NPP v8.3.1 (64 bit)
MX Master 3 w/ Options installed (Application Specific turned off)

**Comment by RealMalWare** [comment]
The horizontal wheel scrolling could be fixed by changing the return value of the WM_MOUSEHWHEEL handler.
If it retruns 1 instead of zero (as microsoft proposes) the scrolling works fine.

This was recently applied in Notepad3
https://github.com/rizonesoft/Notepad3/pull/3988

For reference: there is also a whole article on this topic: https://www.pretentiousname.com/setpoint_hwheel/index.html

**Comment by 00Eraser00** [comment]
Aside from running Notepad++ as admin which made the horizontal scrolling work, I uninstalled Logitech SetPoint which was installed because of an older mouse I think and installed the Logi Options+ Beta which is available [here](https://www.logitech.com/en-us/software/logi-options-plus.html). 
That made horizontal scrolling work without running Notepad++ as admin. Unfortunately, I don't know which of the two things made it work, because I forgot to test it after uninstalling SetPoint.

**Comment by VL4DST3R** [comment]
Will the fix mentioned above by @RealMalWare ever be implemented in np++? Seems to be trivially simple and would help a lot of people.

**Comment by VL4DST3R** [comment]
> > Will the fix mentioned above by @RealMalWare ever be implemented in np++?
> 
> That fix is in Scintilla code. If it is truly a good fix, shouldn't the Scintilla project be the one to integrate it?

If i understand this correctly, it seems to have been fixed only in the version Notepad3 uses, so unless this gets forwarded to the main Scintilla codebase and NP++ subsequently updates to the latest version that incorporates this fix, its on the NP++ team to replicate the changes locally as well.

**Comment by alankilborn** [comment]
The way I understand the philosophy is that things that should be fixed at the Scintilla level should be dealt with there (somewhat obviously).  If Notepad++ were to deal with it, then it has to continually be carried forward, each time Scintilla is updated.  Too much burden for a small thing -- and of course I recognize that if you have this mouse, it is not a small thing to you.  Apparently Notepad3 felt it was worth it to hack its Scintilla.

**Comment by VL4DST3R** [comment]
That is true assuming they DO keep up to update the Scintilla version they use for NP++, which i do not know if it is the case or not. If they don't plan on updating it then fixing this "project-side" like the guys over at Notepad3 did is the only way to address this ticket and others like it. I guess a lead dev should chime in and decide whats the best course of action but i really hope they do end up **addressing** it one way or the other, as its been an issue for years.

**Comment by alankilborn** [comment]
>assuming they DO keep up to update the Scintilla version they use for NP++

The version of Scintilla that Notepad++ uses is updated on an occasional basis; it was updated very recently.  I've manually tracked when Scintilla in N++ gets bumped and here's my list:

    N++ 8.4.1 - Scintilla 5.2.2 and Lexilla 5.1.6
    N++ 8.4   - Scintilla 5.2.1 and Lexilla 5.1.5
    N++ 7.9.4 - Scintilla 4.4.6
    N++ 7.8   - Scintilla 4.2.0
    N++ 7.7   - Scintilla 4.1.4
    N++ 7.6.6 - last to use Scintilla 3.5.6

>I guess a lead dev should chime in and decide whats the best course of action

There's only ONE lead dev, Don Ho.  And obviously this issue hasn't attracted much attention in the past.

**Comment by donho** [comment]
@VL4DST3R 
I cannot reproduce it with my Logitech mouse (no driver installed).
I have done a fix on  #11700, could you check if it solves your problem?

**Comment by RealMalWare** [comment]
Thanks for taking the time. I tested #11700 with [64 bit release](https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/43600715/job/4g6vehlwkn4ek2gq/artifacts) and [32 bit debug](https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/43600715/job/2y3fukreqgocnv56/artifacts)
thumb wheel scrolling works as expected with those (Win 10, Logitech Options installed)

**Comment by marcusthorman** [comment]
@donho Thank you for looking into this - by testing the [x64 release](https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/43600715/job/4g6vehlwkn4ek2gq/artifacts) I can also confirm that https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11700 has completely fixed the issue.
